crates/bevy_tasks/src/task_pool.rs

```diff
@@ -451,6 +484,46 @@ mod tests {
         assert_eq!(count.load(Ordering::Relaxed), 100);
     }
 
+    #[test]
+    fn test_thread_callbacks() {
+        let counter = Arc::new(AtomicI32::new(0));
+        let start_counter = counter.clone();
+        {
+            // Build and immediately drop to terminate
+            TaskPoolBuilder::new()
+                .num_threads(10)
+                .on_thread_spawn(move || {
+                    start_counter.fetch_add(1, Ordering::Relaxed);
+                })
+                .build();
+        }
+        assert_eq!(10, counter.load(Ordering::Relaxed));
+        let end_counter = counter.clone();
+        {
+            TaskPoolBuilder::new()
+                .num_threads(20)
+                .on_thread_destroy(move || {
+                    end_counter.fetch_sub(1, Ordering::Relaxed);
+                })
+                .build();
+        }
+        assert_eq!(-10, counter.load(Ordering::Relaxed));
+        let start_counter = counter.clone();
+        let end_counter = counter.clone();
+        {
+            TaskPoolBuilder::new()
+                .num_threads(5)
+                .on_thread_spawn(move || {
+                    start_counter.fetch_add(1, Ordering::Relaxed);
+                })
+                .on_thread_destroy(move || {
+                    end_counter.fetch_sub(1, Ordering::Relaxed);
+                })
+                .build();
```

Comment:
if it is possible could we add an assert here

Reply:
See other comment.

Reply:
I'm not sure if I'm misunderstanding, but I was saying to do something like this to test that the on_thread_spawn was being called before the threads were destroyed.

```rust
let start_counter = counter.clone();
let end_counter = counter.clone();
let barrier = Arc::new(Barrier::new(6));
let last_barrier = barrier.clone();
{
    let _pool = TaskPoolBuilder::new()
        .num_threads(5)
        .on_thread_spawn(move || {
            start_counter.fetch_add(1, Ordering::Relaxed);
            barrier.clone().wait();
        })
        .on_thread_destroy(move || {
            end_counter.fetch_sub(1, Ordering::Relaxed);
        })
        .build();

    last_barrier.wait();
    assert_eq!(-5, counter.load(Ordering::Relaxed));
}
assert_eq!(-10, counter.load(Ordering::Relaxed));
```

Reply:
Done.